src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -52,11 +52,14 @@ public static function castInterval(\DateInterval $interval, array $a, Stub $stu
 
     private static function formatInterval(\DateInterval $i)
     {
-        $format = '%R '
-            .($i->y ? '%yy ' : '')
-            .($i->m ? '%mm ' : '')
-            .($i->d ? '%dd ' : '')
-        ;
+        $format = '%R ';
+
+        if ($i->m >= 12 || $i->d >= 28 || $i->h >= 24 || $i->i >= 60 || $i->s >= 60) {
+            $i = date_diff($d = new \DateTime(), date_add(clone $d, $i)); // recalculate carry over points
```

Comment:
As months and years are dependant from "now", the only right way is to recalculate carry over points only in 0 month/year case (and with the total number of days as format). This PR will always be useful in most cases, like day interval (``PxD``) and time interval (``PTx``).